src/mlpack/core/data/load_csv.hpp

```diff
@@ -119,33 +205,55 @@ class LoadCSV
     while (std::getline(inFile, line))
     {
       ++rows;
-      // Remove whitespace from either side.
-      boost::trim(line);
-
+      // Remove whitespaces from either side
+      trim(line);
       if (rows == 1)
       {
         // Extract the number of columns.
-        auto findColSize = [&cols](iter_type) { ++cols; };
-        qi::parse(line.begin(), line.end(),
-            stringRule[findColSize] % delimiterRule);
+        std::pair<size_t, size_t> dimen = GetNonNumericMatSize(inFile, delim);
+        cols = dimen.second;
       }
 
       // I guess this is technically a second pass, but that's ok... still the
       // same idea...
       if (MapPolicy::NeedsFirstPass)
       {
         // In this case we must pass everything we parse to the MapPolicy.
-        auto firstPassMap = [&](const iter_type& iter)
+        std::string str(line.begin(), line.end());
+
+        std::stringstream line_stream;
+        std::string token;
+
+        if(line.size() == 0)
+	{
+	  break;
+        }
```

Comment:
This breaks out of the enclosing while loop... why do we want to exit here?  I think this is a little unclear, so if it is a thing we should do, it is probably worth adding a comment here on why the break is needed. :+1:

Reply:
I can't exactly remember why I added that, but I think you are right.
It seems we don't need it here.